### PR TITLE
Refactor the function for generating Gaussian noise

### PR DESCRIPTION
**Pull request by junpenglao**
... that is the same PyTree structure as the input.

relate to #376

### CODE REVIEW

**Review by rlouf — APPROVED** [review]
LGTM

File: blackjax/util.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from jax import jit, lax
+from jax._src.numpy.util import _promote_dtypes
+from jax.flatten_util import ravel_pytree
+from jax.random import normal
+from jax.tree_util import tree_leaves
+
+from blackjax.types import Array, PRNGKey, PyTree
+
+
+@partial(jit, static_argnames=("precision",), inline=True)
```
[the remark this answers is not in the document]
Comment (junpenglao) [reply blackjax/util.py:15]: How about pytree_util.py?

File: blackjax/util.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    if ndim <= 1:
+        return lax.mul(diag_or_dense_a, b)
+    else:
+        return lax.dot(diag_or_dense_a, b, precision=precision)
+
+
+# TODO(https://github.com/blackjax-devs/blackjax/issues/376)
+# Refactor this function to not use ravel_pytree might be more performant.
+def generate_gaussian_noise(
```
[the remark this answers is not in the document]
Comment (junpenglao) [reply blackjax/util.py:56]: +1 to `random_normal`

File: blackjax/sgmcmc/sghmc.py [hunk 3]
```diff
@@ -29,7 +22,7 @@ def one_step(
     ) -> SGLDState:
 
         step, position, logprob_grad = state
-        momentum = sample_momentum(rng_key, position, step_size)
+        momentum = generate_gaussian_noise(rng_key, position, jnp.sqrt(step_size))
```
[the remark this answers is not in the document]
Comment (junpenglao) [reply blackjax/sgmcmc/sghmc.py:25]: I think adding a code comment is sufficient.

**Pull request merged**